### PR TITLE
Implement method program from FlashBuilder

### PR DESCRIPTION
**Pull request by inm-certi**

### CODE REVIEW

File: jocd/src/br/org/certi/jocd/coresight/CortexM.java [hunk 1]
```diff
@@ -1,9 +1,284 @@
...
+  public static final int S_RESET_ST = (1 << 25);
+
+  // Debug Core Register Data Register
+  public static final long DCRDR = 0xE000EDF8;
+
+  // Coprocessor Access Control Register
+  public static final long CPACR = 0xE000ED88;
+  public static final int CPACR_CP10_CP11_MASK = (3 << 20) | (3 << 22);
+
+  public static final long NVIC_AIRCR = (0xE000ED0C);
```
Comment (ghn-certi) [inline jocd/src/br/org/certi/jocd/coresight/CortexM.java]: I don't think parentheses are needed for constants.

File: jocd/src/br/org/certi/jocd/flash/Flash.java [hunk 2]
```diff
@@ -33,6 +35,16 @@
   FlashAlgo flashAlgo;
   Target target;
 
+  boolean flashAlgoDebug = false;
+
+  Integer beginStack;
+  Long beginData;
+  Integer staticBase;
+  int mimProgramLength;
```
Comment (ghn-certi) [inline jocd/src/br/org/certi/jocd/flash/Flash.java]: Typo in the variable name.

File: jocd/src/br/org/certi/jocd/flash/FlashAlgo.java [hunk 3]
```diff
@@ -25,8 +25,8 @@
   public int pcEraseAll;
   public int pcEraseSector;
   public int pcProgramPage;
-  public int beginData;
-  public List<Integer> pageBuffers;
+  public long beginData;
+  public List<Long> pageBuffers;
   public int beginStack;
```
Comment (ghn-certi) [inline jocd/src/br/org/certi/jocd/flash/FlashAlgo.java]: These variables will receive an address. Using int might result in a sign extension error.

File: jocd/src/br/org/certi/jocd/coresight/CortexM.java (hunk 1 above)
Comment (inm-certi) [reply jocd/src/br/org/certi/jocd/coresight/CortexM.java]: Yes, it is not necessary. It is just a literal (and automated) from cortex_m.py.

Fixing...

File: jocd/src/br/org/certi/jocd/flash/Flash.java (hunk 2 above)
Comment (inm-certi) [reply jocd/src/br/org/certi/jocd/flash/Flash.java]: OOps.

File: jocd/src/br/org/certi/jocd/core/MemoryMap.java [hunk 4]
```diff
@@ -15,6 +15,7 @@
  */
 package br.org.certi.jocd.core;
 
+import cz.jaybee.intelhex.Region;
```
[the remark this answers is not in the document]
Comment (inm-certi) [reply jocd/src/br/org/certi/jocd/core/MemoryMap.java]: Fixed

**Pull request merged**